### DISCUSSION TITLE
Allow environment variables to be used as a color

### DISCUSSION THREAD

**Discussion by sigint-run** [JanDeDobbeleer/oh-my-posh#5731] [Ideas]
Similar to my [previous post](https://github.com/JanDeDobbeleer/oh-my-posh/discussions/5360), I suggest being able to use an environment variable as a color.

So I can go from here:
```json
{
  "type": "text",
  "foreground": "#abc123",
  "background": "#123abc",
  "template": "Test"
},
```

to here:
```
{
  "type": "text",
  "foreground": "{{ .Env.VARNAME }}",
  "background": "#123abc",
  "template": "Test"
},
```

with the var name matching a color:
```bash
export VARNAME="#abc123"
```

Of course, the example above only shows the proposition for foreground, but it could work for background as well.
It could also work for color overrides:

```json
{
  "type": "text",
  "foreground": "#abc123",
  "background": "#123abc",
  "template": "Test <{{ .Env.VARNAME }}> with colors!</>"
},
```

Again, the use case is for the same configuration to by dynamic depending on the system where oh-my-posh is used, to theme each systems based on their env variables.

Thanks for reading!

**Comment by JanDeDobbeleer** [comment]
That's why we have `foreground_templates`. This will work there as well.

**Reply by sigint-run** [reply]
It works there? Oh my, I remember testing it and failed, I must have been mistaken. 

Wow, thanks! That will simplify things a lot!

**Reply by sigint-run** [reply]
I've tried the foreground_templates and it works really nice, thank you!

However, color overrides do not work that way (or I have note found how), preventing a single template to use multiple colors with environment variables.

**Reply by JanDeDobbeleer** [reply]
No indeed, it needs to be a valid color. Color overrides are only supported in templates and string values, not color values.
